crates/suiop-cli/dockerfile-templates/typescript.Dockerfile

```diff
@@ -0,0 +1,45 @@
+# this dockerfile was created based on a template that implements the
+# security practices outlined in 
+# https://cheatsheetseries.owasp.org/cheatsheets/NodeJS_Docker_Cheat_Sheet.html#nodejs-docker-cheat-sheet
+
+# we use node as the base image, pinned to a specific hash for reproducibility
+# and to avoid sbom vulnerabilities
+# --> build image
+FROM node:18.17.1-bookworm-slim@sha256:e5c8c319295f6cbc288e19506a9ac37afa3b330f4e38afb01d1269b579cf6a5b AS builder
+# this is the debian:12-slim build for node.
+
+WORKDIR /app
+
+RUN apt -qq update && apt -qq upgrade -y && apt -qq install -y npm && npm install -g typescript
+
+COPY --chown=node:node . /app
+
+COPY --chown=node:node src/ ./src
+COPY --chown=node:node package.json package-lock.json tsconfig.json ./
+COPY --chown=node:node src/ ./src/
+
+RUN npm ci --omit=dev
+RUN npm run build
+
+# same hash as above, but using a separate container to avoid copying our 
```

Comment:
If using `FROM node:18.17.1-bookworm-slim@sha256:e5c8c319295f6cbc288e19506a9ac37afa3b330f4e38afb01d1269b579cf6a5b AS builder as BASE` then in line 27, use `FROM BASE as node-server` or something should be able to avoid copying the whole context?